kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kroxylicious/KroxyliciousApp.java

```diff
@@ -0,0 +1,142 @@
+    private final String clusterIp;
+    private final Thread thread;
+    private long pid;
+
+    /**
+     * Instantiates a new Kroxylicious app.
+     *
+     * @param clusterIp the cluster ip
+     */
+    public KroxyliciousApp(String clusterIp) {
+        this.clusterIp = clusterIp;
+        thread = new Thread(this, "kroxy");
+        thread.start();
+    }
+
+    public void run() {
+        LOGGER.info("Launching kroxylicious app");
+        Path parentPath = Path.of(System.getProperty("user.dir")).getParent();
+        final Path targetPath = parentPath.resolve("kroxylicious-app").resolve("target");
+
+        final Path startScript = resolveStartScript(targetPath);
+        final Path configFile = generateKroxyliciousConfiguration();
+        pid = Exec.execWithoutWait(startScript.toAbsolutePath().toString(), "-c", configFile.toAbsolutePath().toString());
+        while (!thread.isInterrupted()) {
+            try {
+                Thread.sleep(1000);
+            }
+            catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    private Path generateKroxyliciousConfiguration() {
+        try {
+            File configFile = Files.createTempFile("config", ".yaml", TestUtils.getDefaultPosixFilePermissions()).toFile();
+            Files.writeString(configFile.toPath(), KroxyConfigTemplates.getDefaultExternalKroxyConfigMap(clusterIp));
+            configFile.deleteOnExit();
+            return configFile.toPath();
+        }
+        catch (IOException e) {
+            throw new IllegalStateException("Unable to generate kroxylicious configuration file", e);
+        }
+    }
+
+    private static Path resolveStartScript(Path targetPath) {
+        try (Stream<Path> walkStream = Files.walk(targetPath)) {
+            final Optional<Path> startScript = walkStream.filter(Files::isRegularFile).filter(f -> f.endsWith("kroxylicious-start.sh")).findFirst();
+            if (startScript.isEmpty()) {
+                throw new IllegalStateException("unable to find kroxylicious-start.sh");
+            }
+            else {
+                return startScript.get();
+            }
+        }
+        catch (IOException e) {
+            throw new IllegalStateException("unable to find kroxylicious-start.sh", e);
+        }
+    }
+
+    /**
+     * Gets bootstrap.
+     *
+     * @return the bootstrap
+     */
+    public String getBootstrap() {
+        String clusterIP = null;
+        try {
+            var nis = NetworkInterface.getNetworkInterfaces();
+            for (Iterator<NetworkInterface> it = nis.asIterator(); it.hasNext();) {
+                var ni = it.next();
+                for (Iterator<InetAddress> iter = ni.getInetAddresses().asIterator(); iter.hasNext();) {
+                    var i = iter.next();
+                    if (i.getHostAddress().startsWith("10")) {
```

Comment:
This looks very fragile.  What's actually your intent?

Reply:
This was an attempt to connect to Kroxylicious to show that we can launch it without deploying it in kubernetes and then I was trying to get the IP address to connect to, but currently don't use it.